src/WpBookingsShortcode.php

```diff
@@ -1,71 +1,105 @@
+ * on page for booking appointments.
+ *
+ * @since [*next-version*]
+ */
+class WpBookingsShortcode extends AbstractBaseModule
+{
+    /**
+     * The name of shortcode tag.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $shortcodeTag;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable         $key                  The module key.
+     * @param string[]|Stringable[]     $dependencies         The module  dependencies.
+     * @param ContainerFactoryInterface $configFactory        The config factory.
+     * @param ContainerFactoryInterface $containerFactory     The container factory.
+     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     * @param EventManagerInterface     $eventManager         The event manager.
+     * @param EventFactoryInterface     $eventFactory         The event factory.
+     */
+    public function __construct(
+        $key,
+        $dependencies,
+        $configFactory,
+        $containerFactory,
+        $compContainerFactory,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->_initModule($key, $dependencies, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function setup()
+    {
+        return $this->_setupContainer($this->_loadPhpConfigFile(RC_WP_BOOKINGS_SHORTCODE_MODULE_CONFIG), [
+            /*
+             * Transform shortcode parameters before sending them to client.
+             *
+             * @since [*next-version*]
+             */
+            'eddbk_shortcode_parameters_transform_handler' => function (ContainerInterface $c) {
+                return new ShortcodeParametersTransformHandler(
+                    $c->get('eddbk_shortcode/edd_settings/purchase_page'),
+                    $c->get('eddbk_services_select_rm'),
+                    $c->get('sql_expression_builder'),
+                    $c->get('eddbk_admin_edit_services_ui_state_transformer')
+                );
+            },
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        $this->shortcodeTag = $c->get('eddbk_shortcode/shortcode_tag');
+        $wizardBlockFactory = $c->get('eddbk_wizard_block_factory');
+
+        $this->_attach('eddbk_shortcode_parameters_transform', $c->get('eddbk_shortcode_parameters_transform_handler'));
+
+        add_shortcode($this->shortcodeTag, function ($attrs) use ($wizardBlockFactory) {
+            $attrs = $this->_trigger('eddbk_shortcode_parameters', $attrs ? $attrs : [])->getParams();
+            $attrs = $this->_trigger('eddbk_shortcode_parameters_transform', $attrs)->getParams();
```

Comment:
Somebody who writes a handler for this event will not know from its name that the input is shortcode params, while the output must be wizard config.